### ISSUE TITLE
Swipe gestures not detected with libinput 1.18.0

### ISSUE THREAD

**Issue by gschaffner** [bulletmark/libinput-gestures#310]
Hi, and thanks for all the great work!

libinput-gestures isn't picking up on swipe gestures with libinput 1.18.0, but all works with libinput 1.17.3. This is with libinput-gestures 2.67-1 from Manjaro's repos ([PKGBUILD](https://gitlab.manjaro.org/packages/community/libinput-gestures/-/blob/master/PKGBUILD) essentially identical to the one in the AUR).

```
$ libinput-gestures -l
libinput-gestures: session KDE+x11 on Linux-5.10.42-1-MANJARO-x86_64-with-glibc2.33, python 3.9.5, libinput 1.18.0
Hash: 7b225afc6809da81d306436673af8a29
Gestures configured in ~/.config/libinput-gestures.conf:
swipe up         3 _internal --cols 3 ws_up
swipe down       3 _internal --cols 3 ws_down
swipe left       3 _internal --cols 3 ws_left
swipe right      3 _internal --cols 3 ws_right
swipe left_up    3 _internal --cols 3 ws_left_up
swipe left_down  3 _internal --cols 3 ws_left_down
swipe right_up   3 _internal --cols 3 ws_right_up
swipe right_down 3 _internal --cols 3 ws_right_down
swipe up         4 xdotool key XF86AudioPlay
swipe down       4 xdotool key XF86AudioPlay
swipe left       4 xdotool key XF86AudioNext
swipe right      4 xdotool key XF86AudioPrev
swipe_threshold 400
timeout 1.0
libinput-gestures: device /dev/input/by-path/platform-AMDI0010:03-event-mouse(event15): SYNA3290:01 06CB:CD4F Touchpad
libinput-gestures is installed.
libinput-gestures is set up as a desktop application.
libinput-gestures is currently running as a desktop application.
libinput-gestures is set to autostart as a desktop application.
libinput-gestures is using custom configuration file.
```

A quick search and it seems like [libinput 1.18.0](https://lists.freedesktop.org/archives/wayland-devel/2021-June/041849.html) made some gesture changes that may be causing this.

On libinput 1.17.3, all works great:

```
$ libinput-gestures -d
libinput-gestures: session KDE+x11 on Linux-5.10.42-1-MANJARO-x86_64-with-glibc2.33, python 3.9.5, libinput 1.17.3
Hash: 7b225afc6809da81d306436673af8a29
Gestures configured in ~/.config/libinput-gestures.conf:
swipe up         3 _internal --cols 3 ws_up
swipe down       3 _internal --cols 3 ws_down
swipe left       3 _internal --cols 3 ws_left
swipe right      3 _internal --cols 3 ws_right
swipe left_up    3 _internal --cols 3 ws_left_up
swipe left_down  3 _internal --cols 3 ws_left_down
swipe right_up   3 _internal --cols 3 ws_right_up
swipe right_down 3 _internal --cols 3 ws_right_down
swipe up         4 xdotool key XF86AudioPlay
swipe down       4 xdotool key XF86AudioPlay
swipe left       4 xdotool key XF86AudioNext
swipe right      4 xdotool key XF86AudioPrev
swipe_threshold 400
timeout 1.0
libinput-gestures: device /dev/input/by-path/platform-AMDI0010:03-event-mouse(event15): SYNA3290:01 06CB:CD4F Touchpad
libinput-gestures: SWIPE down 3 [-51.4, 645.23]
   _internal --cols 3 ws_down
libinput-gestures: PINCH in 2 [-5.270000000000001, 11.120000000000001]

$ libinput-gestures -r
libinput-gestures: session KDE+x11 on Linux-5.10.42-1-MANJARO-x86_64-with-glibc2.33, python 3.9.5, libinput 1.17.3
Hash: 7b225afc6809da81d306436673af8a29
libinput-gestures: device /dev/input/by-path/platform-AMDI0010:03-event-mouse(event15): SYNA3290:01 06CB:CD4F Touchpad
-event15  DEVICE_ADDED     SYNA3290:01 06CB:CD4F Touchpad    seat0 default group1  cap:pg  size 116x56mm tap(dl off) left scroll-nat scroll-2fg-edge click-buttonareas-clickfinger dwt-on
event15  GESTURE_SWIPE_BEGIN +0.028s    3
event15  GESTURE_SWIPE_UPDATE +0.028s   3 -0.26/12.25 (-1.09/51.40 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.035s   3 -2.69/21.55 (-8.75/69.99 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.043s   3 -2.09/20.91 (-5.47/54.68 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.050s   3 -4.72/27.44 (-12.03/69.99 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.057s   3 -5.21/28.87 (-12.03/66.71 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.064s   3 -4.98/21.59 (-13.12/56.87 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.072s   3 -6.49/38.95 (-14.22/85.30 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.079s   3 -3.53/15.87 (-8.75/39.37 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.086s   3 -2.62/15.71 (-8.75/52.49 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.093s   3 -3.38/19.91 (-9.84/57.96 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.100s   3 -1.64/ 7.86 (-5.47/26.25 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.107s   3 -1.47/ 4.69 (-5.47/17.50 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.114s   3  0.59/ 1.76 ( 2.19/ 6.56 unaccelerated)
event15  GESTURE_SWIPE_END +0.141s      3
event15  GESTURE_PINCH_BEGIN +0.804s    2
event15  GESTURE_PINCH_UPDATE +0.804s   2  2.19/ 0.44 ( 8.20/ 1.64 unaccelerated)  0.95 @  0.12
event15  GESTURE_PINCH_UPDATE +0.811s   2  1.32/ 1.76 ( 4.92/ 6.56 unaccelerated)  0.92 @ -0.19
event15  GESTURE_PINCH_UPDATE +0.818s   2  2.64/ 3.08 ( 9.84/11.48 unaccelerated)  0.86 @ -0.38
event15  GESTURE_PINCH_UPDATE +0.825s   2  0.00/ 6.16 ( 0.00/22.97 unaccelerated)  0.81 @ -0.09
event15  GESTURE_PINCH_UPDATE +0.833s   2  0.00/ 2.64 ( 0.00/ 9.84 unaccelerated)  0.75 @ -0.55
event15  GESTURE_PINCH_UPDATE +0.840s   2  0.88/-1.32 ( 3.28/-4.92 unaccelerated)  0.69 @ -0.56
event15  GESTURE_PINCH_UPDATE +0.847s   2  0.44/ 2.64 ( 1.64/ 9.84 unaccelerated)  0.61 @ -0.24
event15  GESTURE_PINCH_UPDATE +0.854s   2  1.76/ 3.08 ( 6.56/11.48 unaccelerated)  0.54 @ -0.81
event15  GESTURE_PINCH_UPDATE +0.861s   2  4.84/-0.44 (18.04/-1.64 unaccelerated)  0.49 @ -2.78
event15  GESTURE_PINCH_UPDATE +0.869s   2  1.32/ 2.20 ( 4.92/ 8.20 unaccelerated)  0.44 @ -1.16
event15  GESTURE_PINCH_UPDATE +0.876s   2  3.08/ 3.08 (11.48/11.48 unaccelerated)  0.37 @ -1.21
event15  GESTURE_PINCH_UPDATE +0.890s   2  0.88/ 1.32 ( 3.28/ 4.92 unaccelerated)  0.35 @ -0.25
event15  GESTURE_PINCH_UPDATE +0.904s   2  2.20/-0.44 ( 8.20/-1.64 unaccelerated)  0.34 @ -1.62
event15  GESTURE_PINCH_END +0.924s      2

```

But with the same gestures on libinput 1.18.0:

```
$ libinput-gestures -d
libinput-gestures: session KDE+x11 on Linux-5.10.42-1-MANJARO-x86_64-with-glibc2.33, python 3.9.5, libinput 1.18.0
Hash: 7b225afc6809da81d306436673af8a29
Gestures configured in ~/.config/libinput-gestures.conf:
swipe up         3 _internal --cols 3 ws_up
swipe down       3 _internal --cols 3 ws_down
swipe left       3 _internal --cols 3 ws_left
swipe right      3 _internal --cols 3 ws_right
swipe left_up    3 _internal --cols 3 ws_left_up
swipe left_down  3 _internal --cols 3 ws_left_down
swipe right_up   3 _internal --cols 3 ws_right_up
swipe right_down 3 _internal --cols 3 ws_right_down
swipe up         4 xdotool key XF86AudioPlay
swipe down       4 xdotool key XF86AudioPlay
swipe left       4 xdotool key XF86AudioNext
swipe right      4 xdotool key XF86AudioPrev
swipe_threshold 400
timeout 1.0
libinput-gestures: device /dev/input/by-path/platform-AMDI0010:03-event-mouse(event15): SYNA3290:01 06CB:CD4F Touchpad
libinput error: event15 - SYNA3290:01 06CB:CD4F Touchpad: kernel bug: Touch jump detected and discarded.
See https://wayland.freedesktop.org/libinput/doc/1.18.0/touchpad-jumping-cursors.html for details
libinput error: event15 - SYNA3290:01 06CB:CD4F Touchpad: kernel bug: Touch jump detected and discarded.
See https://wayland.freedesktop.org/libinput/doc/1.18.0/touchpad-jumping-cursors.html for details
libinput-gestures: PINCH in 2 [-5.67, 4.580000000000002]
```

Pinches work but swipes don't. The libinput error above also occurs sometimes when I swipe, although not every time. The error also sometimes appears when I didn't swipe but was otherwise using the touchpad. Here's a raw log too:

```
$ libinput-gestures -r
libinput-gestures: session KDE+x11 on Linux-5.10.42-1-MANJARO-x86_64-with-glibc2.33, python 3.9.5, libinput 1.18.0
Hash: 7b225afc6809da81d306436673af8a29
libinput-gestures: device /dev/input/by-path/platform-AMDI0010:03-event-mouse(event15): SYNA3290:01 06CB:CD4F Touchpad
-event15  DEVICE_ADDED     SYNA3290:01 06CB:CD4F Touchpad    seat0 default group1  cap:pg  size 116x56mm tap(dl off) left scroll-nat scroll-2fg-edge click-buttonareas-clickfinger dwt-on
event15  GESTURE_SWIPE_BEGIN +0.036s    3
event15  GESTURE_SWIPE_UPDATE +0.036s   3  0.78/10.42 ( 0.88/11.72 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.043s   3  1.48/11.25 ( 1.46/11.13 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.051s   3  0.84/32.01 ( 0.59/22.26 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.057s   3 -0.51/28.92 (-0.29/16.70 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.064s   3 -0.44/25.38 (-0.29/16.99 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.072s   3 -4.42/46.44 (-2.34/24.60 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.079s   3 -3.93/33.12 (-2.05/17.28 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.086s   3 -5.16/41.30 (-2.93/23.43 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.094s   3 -2.38/23.76 (-1.46/14.64 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.101s   3 -1.28/27.24 (-0.88/18.75 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.108s   3 -1.94/14.36 (-1.46/10.84 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.115s   3  0.59/ 5.04 ( 0.59/ 4.98 unaccelerated)
event15  GESTURE_SWIPE_UPDATE +0.122s   3  0.29/ 3.81 ( 0.29/ 3.81 unaccelerated)
event15  GESTURE_SWIPE_END +0.142s      3
event15  GESTURE_PINCH_BEGIN +0.998s    2
event15  GESTURE_PINCH_UPDATE +0.998s   2  2.64/-3.08 ( 2.64/-3.08 unaccelerated)  0.96 @  1.42
event15  GESTURE_PINCH_UPDATE +1.005s   2 -1.76/-0.44 (-1.76/-0.44 unaccelerated)  0.90 @  1.58
event15  GESTURE_PINCH_UPDATE +1.012s   2 -2.20/ 0.44 (-2.20/ 0.44 unaccelerated)  0.84 @  1.15
event15  GESTURE_PINCH_UPDATE +1.019s   2 -0.44/ 0.44 (-0.44/ 0.44 unaccelerated)  0.80 @  0.74
event15  GESTURE_PINCH_UPDATE +1.027s   2 -0.88/-0.44 (-0.88/-0.44 unaccelerated)  0.74 @  1.18
event15  GESTURE_PINCH_UPDATE +1.034s   2  1.76/-0.44 ( 1.76/-0.44 unaccelerated)  0.66 @  1.26
event15  GESTURE_PINCH_UPDATE +1.041s   2  0.00/ 0.00 ( 0.00/ 0.00 unaccelerated)  0.62 @  0.42
event15  GESTURE_PINCH_UPDATE +1.048s   2 -2.64/-1.32 (-2.64/-1.32 unaccelerated)  0.55 @  1.36
event15  GESTURE_PINCH_UPDATE +1.056s   2 -0.44/-0.88 (-0.44/-0.88 unaccelerated)  0.48 @  1.87
event15  GESTURE_PINCH_UPDATE +1.063s   2  1.76/-1.32 ( 1.76/-1.32 unaccelerated)  0.45 @  0.99
event15  GESTURE_PINCH_UPDATE +1.070s   2  0.00/-0.88 ( 0.00/-0.88 unaccelerated)  0.40 @  2.14
event15  GESTURE_PINCH_UPDATE +1.077s   2  0.44/-2.64 ( 0.44/-2.64 unaccelerated)  0.36 @  3.09
event15  GESTURE_PINCH_UPDATE +1.084s   2 -1.32/-2.64 (-1.32/-2.64 unaccelerated)  0.33 @  4.35
event15  GESTURE_PINCH_UPDATE +1.092s   2 -0.44/-3.52 (-0.44/-3.51 unaccelerated)  0.32 @  3.32
event15  GESTURE_PINCH_UPDATE +1.099s   2 -1.76/ 0.00 (-1.76/ 0.00 unaccelerated)  0.32 @  0.48
event15  GESTURE_PINCH_END +1.133s      2
```

Thanks again for all the great work on this tool.

**Comment by bulletmark** [comment]
I'm on Arch where libinput updated to 1.18.0-1 a month ago today but I have not seen any problems. Swipe gestures still work fine for me.

Ignore that "kernel bug: Touch jump detected and discarded" message from libinput. It has been seen by very many users, including myself and I suspect everybody, for many years and is completely benign. It will be happening for you also even with libinput 1.17 and earlier.

So in simple terms, please confirm that when you run  `libinput-gestures -d` and you do all types of swipe gesture then you see nothing reported? And when you do pinch gestures they are reported?

**Comment by gschaffner** [comment]
Sorry for the delay. That is correct. On libinput 1.17.3, all 8 swipes show up in `libinput-gestures -d`, and pinch in/out show up too. On 1.18.0-1, only pinch in/out show up in `libinput-gestures -r`, even though swipes show up in `libinput-gestures -r`.

(I did just notice that the rotation pinches don't work on my machine, but they aren't reported on either libinput 1.17.3 or 1.18.0.)

I don't think that this could be due to anything other than the libinput update, as swipes work again immediately upon downgrading with `sudo pacman -U /var/cache/pacman/pkg/libinput-1.17.3-1-x86_64.pkg.tar.zst`.

**Comment by bulletmark** [comment]
Sorry but I asked about `libinput-gestures -d` (not `-r`). Please confirm whether swipes and pinches work with that command. Please test in old and new libinput versions.

I doubt that anything fundamental has changed with libinput 1.18 because it still works fine for me. Do you notice any other differences with that version and your touchpad?

**Comment by gschaffner** [comment]
Sorry, I made a typo in the above comment. Changed `-r` to `-d`.

I haven't noticed any other problems with my touchpad on 1.18.0.

**Comment by bulletmark** [comment]
OK, you are saying swipes appear with `-r` but not with `-d`. Please run `libinput-gestures -r`, do some swipes, and paste the output here.

**Comment by gschaffner** [comment]
```
$ libinput-gestures -r
libinput-gestures: session KDE+x11 on Linux-5.10.42-1-MANJARO-x86_64-with-glibc2.33, python 3.9.5, libinput 1.18.0
Hash: 7b225afc6809da81d306436673af8a29
libinput-gestures: device /dev/input/by-path/platform-AMDI0010:03-event-mouse(event6): SYNA3290:01 06CB:CD4F Touchpad
-event6   DEVICE_ADDED     SYNA3290:01 06CB:CD4F Touchpad    seat0 default group1  cap:pg  size 116x56mm tap(dl off) left scroll-nat scroll-2fg-edge click-buttonareas-clickfinger dwt-on
event6   GESTURE_SWIPE_BEGIN +0.036s    3
event6   GESTURE_SWIPE_UPDATE +0.036s   3 -0.26/ 9.64 (-0.29/10.84 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.043s   3 -0.62/15.82 (-0.59/14.94 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.050s   3  0.00/ 9.53 ( 0.00/ 8.49 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.058s   3  0.00/21.36 ( 0.00/18.16 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.065s   3  0.41/20.38 ( 0.29/14.64 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.072s   3  0.69/12.69 ( 0.59/10.84 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.079s   3  1.65/15.51 ( 1.46/13.77 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.086s   3  0.77/22.00 ( 0.59/16.70 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.093s   3  1.42/12.39 ( 1.17/10.25 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.100s   3  1.87/12.77 ( 1.76/12.01 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.108s   3  1.08/20.11 ( 0.88/16.40 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.115s   3 -0.35/13.65 (-0.29/11.42 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.121s   3  0.00/ 7.72 ( 0.00/ 7.62 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.129s   3  0.88/ 7.92 ( 0.88/ 7.91 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.136s   3 -0.29/ 7.92 (-0.29/ 7.91 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.143s   3  0.00/ 2.93 ( 0.00/ 2.93 unaccelerated)
event6   GESTURE_SWIPE_END +0.170s      3
event6   GESTURE_SWIPE_BEGIN +0.602s    3
event6   GESTURE_SWIPE_UPDATE +0.602s   3 20.63/-3.24 (20.50/-3.22 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.609s   3 27.99/-1.20 (20.50/-0.88 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.616s   3 50.45/ 1.22 (24.31/ 0.59 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.623s   3 58.03/-0.69 (24.60/-0.29 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.630s   3 68.89/ 4.49 (26.95/ 1.76 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.638s   3 94.21/ 0.89 (31.05/ 0.29 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.645s   3 86.98/ 2.75 (27.83/ 0.88 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.652s   3 85.43/ 6.23 (28.12/ 2.05 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.659s   3 84.95/ 3.50 (28.41/ 1.17 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.666s   3 32.88/ 4.18 (16.11/ 2.05 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +0.673s   3  6.87/ 2.29 ( 6.15/ 2.05 unaccelerated)
event6   GESTURE_SWIPE_END +0.715s      3
event6   GESTURE_SWIPE_BEGIN +1.339s    3
event6   GESTURE_SWIPE_UPDATE +1.339s   3 15.83/-13.48 (15.82/-13.47 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.346s   3 22.07/-21.67 (16.11/-15.82 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.354s   3 34.59/-27.55 (17.28/-13.77 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.361s   3 40.90/-34.51 (18.75/-15.82 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.368s   3 44.83/-33.29 (19.33/-14.35 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.375s   3 53.22/-43.48 (20.80/-16.99 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.382s   3 53.98/-53.15 (19.04/-18.75 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.389s   3 34.85/-30.25 (15.52/-13.47 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.397s   3 49.51/-40.69 (21.38/-17.57 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.404s   3 40.86/-43.09 (16.11/-16.99 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.411s   3 34.12/-28.13 (16.70/-13.77 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.418s   3 39.54/-29.65 (18.75/-14.06 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.425s   3 11.81/-13.70 ( 7.32/-8.49 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.432s   3 26.14/-26.59 (16.70/-16.99 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.439s   3 12.26/-11.35 ( 7.91/-7.32 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.446s   3  2.95/-2.36 ( 2.93/-2.34 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +1.453s   3  1.76/-1.76 ( 1.76/-1.76 unaccelerated)
event6   GESTURE_SWIPE_END +1.481s      3
event6   GESTURE_PINCH_BEGIN +2.076s    2
event6   GESTURE_PINCH_UPDATE +2.076s   2  0.00/-0.40 ( 0.00/-0.44 unaccelerated)  0.95 @ -0.14
event6   GESTURE_PINCH_UPDATE +2.083s   2  1.60/-1.20 ( 1.76/-1.32 unaccelerated)  0.88 @ -0.49
event6   GESTURE_PINCH_UPDATE +2.091s   2  0.44/ 2.64 ( 0.44/ 2.64 unaccelerated)  0.82 @ -0.48
event6   GESTURE_PINCH_UPDATE +2.097s   2  0.00/ 3.52 ( 0.00/ 3.51 unaccelerated)  0.75 @  0.71
event6   GESTURE_PINCH_UPDATE +2.105s   2  3.52/-1.32 ( 3.51/-1.32 unaccelerated)  0.68 @  0.38
event6   GESTURE_PINCH_UPDATE +2.112s   2  5.28/-5.28 ( 5.27/-5.27 unaccelerated)  0.62 @ -0.74
event6   GESTURE_PINCH_UPDATE +2.119s   2  2.20/ 0.00 ( 2.20/ 0.00 unaccelerated)  0.56 @ -0.62
event6   GESTURE_PINCH_UPDATE +2.127s   2  1.32/ 1.76 ( 1.32/ 1.76 unaccelerated)  0.52 @  0.15
event6   GESTURE_PINCH_UPDATE +2.134s   2  3.96/ 1.76 ( 3.95/ 1.76 unaccelerated)  0.47 @ -0.37
event6   GESTURE_PINCH_UPDATE +2.141s   2  3.52/-0.88 ( 3.51/-0.88 unaccelerated)  0.42 @ -0.56
event6   GESTURE_PINCH_UPDATE +2.148s   2  3.08/-0.88 ( 3.08/-0.88 unaccelerated)  0.40 @ -0.85
event6   GESTURE_PINCH_UPDATE +2.155s   2  1.32/ 0.00 ( 1.32/ 0.00 unaccelerated)  0.39 @ -0.43
event6   GESTURE_PINCH_UPDATE +2.162s   2  1.76/ 1.32 ( 1.76/ 1.32 unaccelerated)  0.38 @  0.01
event6   GESTURE_PINCH_UPDATE +2.169s   2  0.88/ 0.00 ( 0.88/ 0.00 unaccelerated)  0.38 @ -0.55
event6   GESTURE_PINCH_END +2.196s      2
event6   GESTURE_SWIPE_BEGIN +2.700s    3
event6   GESTURE_SWIPE_UPDATE +2.700s   3  0.59/11.43 ( 0.59/11.42 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.707s   3  0.65/18.94 ( 0.59/16.99 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.715s   3  0.49/35.99 ( 0.29/21.67 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.721s   3  1.29/19.40 ( 0.88/13.18 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.729s   3  0.00/26.82 ( 0.00/19.04 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.736s   3 -1.43/30.03 (-0.88/18.45 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.743s   3 -0.42/20.44 (-0.29/14.35 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.750s   3 -1.41/34.78 (-0.88/21.67 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.757s   3 -0.45/20.47 (-0.29/13.18 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.765s   3  0.00/20.55 ( 0.00/16.70 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.772s   3  0.45/28.60 ( 0.29/18.75 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.778s   3  0.00/15.07 ( 0.00/10.84 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.786s   3  0.60/ 8.64 ( 0.59/ 8.49 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.793s   3  0.59/10.93 ( 0.59/10.84 unaccelerated)
event6   GESTURE_SWIPE_UPDATE +2.800s   3 -0.91/10.58 (-0.88/10.25 unaccelerated)
event6   GESTURE_SWIPE_END +2.819s      3
```

**Comment by bulletmark** [comment]
The problem is due to that `swipe_threshold 400` you added in your config. Get rid of that and it works. Why do you need that?

**Comment by gschaffner** [comment]
That seems to be it, thanks!

As to why I had `swipe_threshold 400` in my config:

```
# don't act on swipes shorter than n dots
# my touchpad is approx. 1240x900 dots
swipe_threshold 400
```

I guess the question now is why `swipe_threshold 400` breaks swipes on libinput 1.18.0 but works fine on libinput 1.17.3. Maybe this is due to the scaling changes introduced for deltas in https://gitlab.freedesktop.org/libinput/libinput/-/commit/60d5172e15728cc25db889a7a6bcf37a06a15a3a? If I am reading the `libinput-gestures -r` outputs correctly (I think the parenthetical bits are deltas?), I am seeing much smaller deltas on 1.18.0 than on 1.17.3.

**Comment by bulletmark** [comment]
No idea why libinput changed that but is not a bug in libinput-gestures so closing this.
